mypyc/irbuild/statement.py

```diff
@@ -930,16 +932,35 @@ def emit_yield_from_or_await(
     to_yield_reg = Register(object_rprimitive)
     received_reg = Register(object_rprimitive)
 
-    get_op = coro_op if is_await else iter_op
-    if isinstance(get_op, PrimitiveDescription):
-        iter_val = builder.primitive_op(get_op, [val], line)
+    helper_method = "__mypyc_generator_helper__"
+    if (
+        isinstance(val, (Call, MethodCall))
+        and isinstance(val.type, RInstance)
+        and val.type.class_ir.has_method(helper_method)
+    ):
+        # This is a generated generator class, and we can use a fast path.
+        iter_val: Value = val
```

Comment:
The changes in this function implemented the bulk of the optimization.

Reply:
Maybe add a bit more detailed comment (essentially summarizing the PR description) so we will not forget the motivation.